app/products/product_info.py

```diff
@@ -22,7 +22,24 @@ def create_match_paterns(self):
     def dispose(self):
         super().dispose()
 
-    def handle(self, message: str) -> str:
+    def handle_prod_intent(self, product: str, intent: str) -> str:
+
+        request = None
+
+        for prod in MOCK_PRODUCT_DATA:
```

Comment:
Please use `execute_query()` to search for product. This method does a linear scan, which is very expensive,